docs/docs/vscode-setup.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Setting up Visual Studio Code for Gatsby development"
+---
+
+VS Code is a wonderful tool that can require some initial setup to get everything working. Here's some recommendations to get code flowing smoothly.
+
+## Initial Setup
+
+### .gitignore
+
+You should add the `.vscode` folder to your `.gitignore`, as your workspace settings and perferences should not be committed to a git repository. Consider setting up an `.editorconfig` file if you need to sync different settings like linting preferences across editors.
+
+### Correcting IntelliSense
+
+You should add the `typescript` package as a global or dev dependency for your project so that you can benefit from the TypeScript-driven refactoring and other development goodies.
+
+`yarn global add typescript` will add it globally.
+
+The following `jsconfig.json` is reccomended for getting better IntelliSense on your `.js` files while working.
+
+```json5
+{
+  "compilerOptions": {
+    "target": "es2016",
+    "jsx": "react",
+    // "baseUrl": "src", if you are using gatsby-plugin-resolve-src
+    "moduleResolution": "node"
+  },
+  "include": ["src"] // Don't pickup on .cache or public by accident!
+}
+```
```

Comment:
Likewise, I don't use this config and everything seems to work okay for me out of the box...

Reply:
For me, I was getting my plugin, .cache, and public folder and [it was a right mess](https://cdn.discordapp.com/attachments/154373094415269888/485544524568068096/unknown.png).

I'm also on a windows box with my own myriad of system quirks. I do find things end up going a lot smoother on Linux comparatively.